File: 0246. Strobogrammatic Number/Solution0246.cpp
class Solution0246 {
public:
    bool isStrobogrammatic(string num) {
        unordered_map<char, char> m{{'0','0'}, {'1','1'}, {'6','9'}, {'8','8'}, {'9','6'}};
        int left = 0, right = num.length() - 1;
        while (left <= right) {
            if (m[num[left++]] != num[right--]) return false;
        }
        return true;        
    }
};

File: 0246. Strobogrammatic Number/Solution0246.java
public class Solution0246 {

    Map<Character, Character> dict = new HashMap<>(
    Map.of('0', '0', '1', '1', '6', '9', '8',  '8', '9', '6'));
    
    public boolean isStrobogrammatic(String num) {
        int i = 0;
        int j = num.length() - 1;
        while (i < j) {
            char left = num.charAt(i++);
            char right = num.charAt(j--);
            if (!dict.containsKey(left) || dict.get(left) != right) {
                return false;
            }
        }
        if (i != j) {
            return true;
        }
        return num.charAt(i) == '0' || num.charAt(i) == '1' || num.charAt(i) == '8';
    }
  
}
